### DISCUSSION TITLE
Additional energy term in TotalFreeEnergy

### DISCUSSION THREAD

**Discussion by rnpratoori** [idaholab/moose#28369] [Q&A Modules: Phase field]
Hello team,

I need to calculate the total energy of the system for a phase separation in diblock copolymers. The energy function has an additional term compared to what is described [here](https://mooseframework.inl.gov/modules/phase_field/Phase_Field_Equations.html). I feel utilizing the `additional_free_energy` option in `TotalFreeEnergy` is the easiest way.

The additional term is of the form $$\int_\Omega G(x,y)(u(x)-m)dy$$. Here, both $$x$$ and $$y$$ are the position vectors not just coordinates. $$x$$ is the Quadrature point at which the integral has to be calculated to contribute to the free energy.

Now, how can this be written most efficiently?

**Comment by GiudGiud** [comment]
Hello

So this integral is being computed over the whole volume?
And are you missing a dx in there? Or is the "x" the currentl location (e.g. this term is a field not a single value)

Guillaume

**Reply by rnpratoori** [reply]
@GiudGiud Sorry for the confusion. I edited the expression. The integral is part of another integral (also over the whole domain) which is with dx. 

Yes, the integral is over the whole volume.
Yes, the "x" is the current location. So, this integral has to be calculated at each quadrature point ("x") for the whole volume.

**Reply by GiudGiud** [reply]
So it seems we need some sort of sweeping step to compute this integral over the entire mesh.
@dschwen @laagesen @jessecarterMOOSE

**Comment by laagesen** [comment]
I don't think we have a way to do this currently, unless there is something similar from the Rigid Body Motion work that was done several years ago, can you comment on this @SudiptaBiswas?
